docs/BUILD.rst

```diff
@@ -0,0 +1,86 @@
+##########
+Lago build
+##########
+
+Lago allows to build / configure VM disks during init stage.
+In the init file, the key ``build`` should be added to each disk that needs to be configured.
+
+``build`` should map to a list of Builders, where each builder in the list is
+a one entry dictionary that maps to a dictionary that holds the options for that builder.
+
+Options are key-value pairs, where the key is the name of the option
+(without leading dashes), and the value is the argument for that option.
+If the option takes no arguments, the empty string should be set as the value.
+If the builder allows specifying an options multiple times, the value should
+be a list of arguments.
+
+
+Builders
+===========
+
+Builders are commands that can be used to build/configure VMs.
+Builders are called by the order they appear in the init file.
+
+virt-customize
+---------------
+A tool for customizing a virtual machine (install packages, copying files, etc...).
+`virt-customize` is part of the `libguestfs` tool set which is part of Lago's dependencies.
+
+`virt-customize` can be called only on disks which contains an OS.
+
+Depends on the version of `virt-customize` installed on your system (it can vary between
+different OS), all the valid options for `virt-customize` can be specified in the init file.
+For the full list of options please refer to `virt-customize documentation`_.
+
+There is a special case when using `virt-customize` to inject ssh keys. If the
+empty string is provided to ``ssh-inject`` option, Lago will replace it with
+the path to the environment's ssh key.
```

Comment:
maybe /to the environment's ssh key/lago's self generated ssh keys/g to make it obvious it is not the user's own ssh keys.